src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        LocalDate firstDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate secondDate = LocalDate.parse(dateTo, formatter);
+        for (String name: names) {
+            int salary = 0;
+            for (String dataLine: data) {
+                String[] elementOfData = dataLine.split(" ");
+                LocalDate date = LocalDate.parse(elementOfData[0], formatter);
+                if ((date.isAfter(firstDate) || date.equals(firstDate))
+                        && (date.isBefore(secondDate) || date.equals(secondDate))
+                        && elementOfData[1].equals(name)) {
+                    salary += Integer.parseInt(elementOfData[2])
+                            * Integer.parseInt(elementOfData[3]);
+                }
+            }
+            stringBuilder.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+        return stringBuilder.toString();
     }
```

Comment:
This method implementation doesn't contain any empty lines at the beginning, which is good. However, be cautious with empty lines inside the method as well. Make sure to remove any that are not necessary for readability or logical separation.